src/decoder/lossless.rs

```diff
@@ -0,0 +1,212 @@
+use decoder::{Decoder, MAX_COMPONENTS};
+use error::{Error, Result, UnsupportedFeature};
+use huffman::{fill_default_mjpeg_tables, HuffmanDecoder, HuffmanTable};
+use marker::Marker;
+use parser::Predictor;
+use parser::{
+    parse_app, parse_com, parse_dht, parse_dqt, parse_dri, parse_sof, parse_sos,
+    AdobeColorTransform, AppData, CodingProcess, Component, Dimensions, EntropyCoding, FrameInfo,
+    IccChunk, ScanInfo,
+};
+use std::io::Read;
+
+impl<R: Read> Decoder<R> {
+    /// decode_scan_lossless
+    pub fn decode_scan_lossless(
+        &mut self,
+        frame: &FrameInfo,
+        scan: &ScanInfo,
+    ) -> Result<(Option<Marker>, Option<Vec<Vec<u16>>>)> {
+        let ncomp = scan.component_indices.len();
+        assert!(ncomp <= MAX_COMPONENTS);
+        let mut results = vec![Vec::new(); MAX_COMPONENTS];
```

Comment:
Should this not have a length of `scan.component_indices.len()`?

Reply:
right